src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
   // write your code here
+  const styles = {};
+  const properties = sourceString.split(';\n');
+
+  properties.forEach(property => {
+    const [name, value] = property
+      .split(':')
+      .map(part => part.trim());
+
+    if (name !== '') {
```

Comment:
The if-statement checks if the name is not an empty string, but it does not check if the value is not undefined. If a style property has no value, this could introduce undefined values into your styles object.